malaria-ios/Managers/NotificationManager.swift

```diff
@@ -3,52 +3,76 @@ import UIKit
-        super.init(context: context)
+    guard let data = localNotificationArrayData else {
+      return
     }
     
-    /// Schedule a notification at the fireTime given by argument
-    ///
-    /// All previous notifications will be unsheduled
-    ///
-    /// - parameter `NSDate`:: fireTime
-    public func scheduleNotification(fireTime: NSDate){
-        Logger.Info("Sheduling \(category) to " + fireTime.formatWith("dd-MMMM-yyyy hh:mm"))
-        let notification: UILocalNotification = createNotification(fireTime)
-        UIApplication.sharedApplication().scheduleLocalNotification(notification)
-    }
+    let scheduledNotifications = NSKeyedUnarchiver.unarchiveObjectWithData(data) as? [UILocalNotification]
     
-    /// Unschedule all notifications with the category
-    public func unsheduleNotification(){
-        for event in UIApplication.sharedApplication().scheduledLocalNotifications! {
-            if event.category == category {
-                UIApplication.sharedApplication().cancelLocalNotification(event)
-            }
-        }
+    guard let notifications = scheduledNotifications else {
+      return
     }
     
-    /// Creates a notification
-    ///
-    /// - parameter `NSDate`:: fireTime
-    /// - returns: `UILocalNotification`: local notification
-    private func createNotification(fireDate: NSDate) -> UILocalNotification{
-        let localNotification = UILocalNotification()
-        localNotification.fireDate = fireDate
-        localNotification.soundName = UILocalNotificationDefaultSoundName;
-        localNotification.alertAction = alertAction
-        localNotification.alertBody = alertBody
-        localNotification.soundName = UILocalNotificationDefaultSoundName
-        localNotification.category = category
-        
-        return localNotification
+    for notification in notifications
+      where notification.category == category {
+        UIApplication.sharedApplication().cancelLocalNotification(notification)
     }
+  }
+  
+  /// Creates a notification
+  ///
+  /// - parameter `NSDate`:: fireTime
+  /// - returns: `UILocalNotification`: local notification
+  private func createNotification(fireDate: NSDate) -> UILocalNotification{
+    let localNotification = UILocalNotification()
+    localNotification.fireDate = fireDate
+    localNotification.soundName = UILocalNotificationDefaultSoundName;
+    localNotification.alertAction = alertAction
+    localNotification.alertBody = alertBody
+    localNotification.soundName = UILocalNotificationDefaultSoundName
+    localNotification.category = category
+    
+    scheduledNotifications.append(localNotification)
+    
+    // Save the notifications persistently
+    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
+    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
```

Comment:
Another example... imagine if NSUserDefaults changes internally so that
custom objects are only stored withing specific conditions. With those unit
tests, you will be able to tackle the issue more rapidly.

2016-06-25 21:15 GMT+01:00 Teodor Ciuraru <notifications@github.com>:

> In malaria-ios/Managers/NotificationManager.swift
> <https://github.com/systers/malaria-app-ios/pull/52#discussion_r68493345>:
>
> > +  /// - parameter `NSDate`:: fireTime
> > +  /// - returns: `UILocalNotification`: local notification
> > +  private func createNotification(fireDate: NSDate) -> UILocalNotification{
> > +    let localNotification = UILocalNotification()
> > +    localNotification.fireDate = fireDate
> > +    localNotification.soundName = UILocalNotificationDefaultSoundName;
> > +    localNotification.alertAction = alertAction
> > +    localNotification.alertBody = alertBody
> > +    localNotification.soundName = UILocalNotificationDefaultSoundName
> > +    localNotification.category = category
> > +
> > +    scheduledNotifications.append(localNotification)
> > +
> > +    // Save the notifications persistently
> > +    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
> > +    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
>
> Sorry, I don't understand. Apple guarantees the fact that NSUserDefaults
> saves NSObjects that can be later retrieved. Do you have any special case
> in mind?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/systers/malaria-app-ios/pull/52/files/1bd98847e91e9af59711c1205e2643b2daf28fd7#r68493345>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEgjsZ1npQnuuIioBAUz8lGZiGFug_eoks5qPYxlgaJpZM4I-I45>
> .
>



-- 
Os melhores cumprimentos,
Bruno Henriques